plugin/firewall/firewall.go

```diff
@@ -0,0 +1,125 @@
+// Package firewall enables filtering on query and response using direct expression as policy.
+// it allows interact with other Policy Engines if those are plugin implementing the Engineer interface
+package firewall
+
+import (
+	"context"
+	"errors"
+
+	"github.com/coredns/coredns/plugin/pkg/dnstest"
+
+	"github.com/coredns/coredns/plugin/pkg/policy"
+
+	"github.com/coredns/coredns/plugin"
+	"github.com/coredns/coredns/plugin/pkg/nonwriter"
+
+	"github.com/coredns/coredns/request"
+	"github.com/miekg/dns"
+)
+
+var (
+	errInvalidAction = errors.New("invalid action")
+)
+
+// firewallPlugin represents a plugin instance that can validate DNS
+// requests and replies using rulelists on the query and/or on the reply
+type firewallPlugin struct {
+	engines map[string]policy.Engine
+	query   *ruleList
+	reply   *ruleList
+
+	next plugin.Handler
+}
+
+func newFirewallPlugin() (*firewallPlugin, error) {
+	pol := &firewallPlugin{}
+	pol.engines = make(map[string]policy.Engine, 0)
+	pol.engines["--default--"] = policy.NewExpressionEngine()
```

Comment:
these 3 lines can be just 1 line does initialize the plugin and sets the default.

Reply:
ok